src/typing_extensions.py

```diff
@@ -315,6 +316,33 @@ class Other(Leaf):  # Error reported by type checker
         return f
 
 
+if hasattr(typing, "solid_base"):
```

Comment:
```suggestion
if hasattr(typing, "solid_base"):  # 3.15
```

To more easily remove the conditional when dropping support for Python versions.

Reply:
That's optimistic :)